x-pack/plugins/maps/public/routing/routes/maps_app/maps_app_view.js

```diff
@@ -0,0 +1,486 @@
+        mapInitLayers = mapInitLayers.substr(0, mapInitLayers.length - 1);
+      }
+      return rison.decode_array(mapInitLayers);
+    } catch (e) {
+      getToasts().addWarning({
+        title: i18n.translate('xpack.maps.initialLayers.unableToParseTitle', {
+          defaultMessage: `Initial layers not added to map`,
+        }),
+        text: i18n.translate('xpack.maps.initialLayers.unableToParseMessage', {
+          defaultMessage: `Unable to parse contents of 'initialLayers' parameter. Error: {errorMsg}`,
+          values: { errorMsg: e.message },
+        }),
+      });
+      return [];
+    }
+  }
+
+  async _updateIndexPatterns(nextIndexPatternIds) {
+    const indexPatterns = [];
+    const getIndexPatternPromises = nextIndexPatternIds.map(async (indexPatternId) => {
+      try {
+        const indexPattern = await getIndexPatternService().get(indexPatternId);
+        indexPatterns.push(indexPattern);
+      } catch (err) {
+        // unable to fetch index pattern
+      }
+    });
+
+    await Promise.all(getIndexPatternPromises);
+    this.setState({
+      indexPatterns,
+    });
+  }
+
+  _handleStoreChanges = () => {
+    const { prevIndexPatternIds } = this.state;
+    const { nextIndexPatternIds } = this.props;
+
+    if (nextIndexPatternIds !== prevIndexPatternIds) {
+      this.setState({ prevIndexPatternIds: nextIndexPatternIds });
+      this._updateIndexPatterns(nextIndexPatternIds);
+    }
+  };
+
+  _getAppStateFilters = () => {
+    return this._appStateManager.getFilters() || [];
+  };
+
+  _syncAppAndGlobalState = () => {
+    const { query, time, refreshConfig, initialized } = this.state;
+    const { filterManager } = getData().query;
+
+    // appState
+    this._appStateManager.setQueryAndFilters({
+      query: query,
+      filters: filterManager.getAppFilters(),
+    });
+
+    // globalState
+    const refreshInterval = {
+      pause: refreshConfig.isPaused,
+      value: refreshConfig.interval,
+    };
+    updateGlobalState(
+      {
+        time: time,
+        refreshInterval,
+        filters: filterManager.getGlobalFilters(),
+      },
+      !initialized
+    );
+    this.setState({ refreshInterval });
+  };
+
+  _onQueryChange = async ({ filters, query, time, refresh }) => {
+    const { filterManager } = getData().query;
+    const { dispatchSetQuery } = this.props;
+    const newState = {};
+    let newFilters;
+    if (filters) {
+      filterManager.setFilters(filters); // Maps and merges filters
+      newFilters = filterManager.getFilters();
+      newState.filters = newFilters;
+    }
+    if (query) {
+      newState.query = query;
+    }
+    if (time) {
+      newState.time = time;
+    }
+    this.setState(newState, () => {
+      this._syncAppAndGlobalState();
+      dispatchSetQuery(
+        refresh,
+        newFilters || this.state.filters,
+        query || this.state.query,
+        time || this.state.time
+      );
+      updateGlobalState(
```

Comment:
Not sure if you need this call to `updateGlobalState` since `this._syncAppAndGlobalState` also calls `updateGlobalState`